drivers/sensor/wsen_pads/wsen_pads_trigger.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2023 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT we_wsen_pads
+
+#include <zephyr/logging/log.h>
+
+#include "wsen_pads.h"
+
+LOG_MODULE_DECLARE(WSEN_PADS, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Enable/disable data-ready interrupt handling */
+static inline int pads_setup_drdy_interrupt(const struct device *dev, bool enable)
+{
+	const struct pads_config *cfg = dev->config;
+	unsigned int flags = enable ? GPIO_INT_EDGE_TO_ACTIVE : GPIO_INT_DISABLE;
+
+	return gpio_pin_interrupt_configure_dt(&cfg->gpio_drdy, flags);
+}
+
+/*
+ * Is called when a data-ready interrupt has occurred. Triggers
+ * asynchronous processing of the interrupt in pads_process_drdy_interrupt().
+ */
+static inline void pads_handle_drdy_interrupt(const struct device *dev)
+{
+	struct pads_data *data = dev->data;
+
+	/* Disable interrupt handling until the interrupt has been processed */
+	pads_setup_drdy_interrupt(dev, false);
+
+#if defined(CONFIG_WSEN_PADS_TRIGGER_OWN_THREAD)
+	k_sem_give(&data->drdy_sem);
+#elif defined(CONFIG_WSEN_PADS_TRIGGER_GLOBAL_THREAD)
+	k_work_submit(&data->work);
+#endif
+}
+
+/* Calls data-ready trigger handler (if any) */
+static void pads_process_drdy_interrupt(const struct device *dev)
+{
+	struct pads_data *data = dev->data;
+
+	if (data->data_ready_handler != NULL) {
+		data->data_ready_handler(dev, &data->data_ready_trigger);
+		pads_setup_drdy_interrupt(dev, true);
+	}
+}
+
+int pads_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
+		     sensor_trigger_handler_t handler)
+{
+	struct pads_data *data = dev->data;
+	const struct pads_config *cfg = dev->config;
+	int32_t pressure_dummy;
+
+	if (trig->type != SENSOR_TRIG_DATA_READY) {
+		LOG_ERR("Unsupported sensor trigger");
+		return -ENOTSUP;
+	}
+
+	pads_setup_drdy_interrupt(dev, false);
+
+	data->data_ready_handler = handler;
+	if (handler == NULL) {
+		/* Disable data-ready interrupt */
+		if (PADS_enableDataReadyInterrupt(&data->sensor_interface, PADS_disable) !=
+		    WE_SUCCESS) {
+			LOG_ERR("Failed to disable data-ready interrupt.");
+			return -EIO;
+		}
+		return 0;
+	}
+
+	data->data_ready_trigger = *trig;
```

Comment:
Store the pointer instead of making a copy of the struct. This will allow the application to use `CONTAINER_OF` in the handler